code/datums/helper_datums/teleport.dm

```diff
@@ -180,3 +180,36 @@
 
 /proc/get_teleport_turf(turf/center, precision = 0)
 	return safepick(get_teleport_turfs(center, precision))
+
+/proc/wizarditis_teleport(var/mob/living/carbon/affected_mob)
+	var/list/theareas = get_areas_in_range(80, affected_mob)
+	for(var/area/space/S in theareas)
+		theareas -= S
+
+	if(!theareas||!theareas.len)
+		return
+
+	var/area/thearea = pick(theareas)
+
+	var/list/L = list()
+	for(var/turf/T in get_area_turfs(thearea.type))
+		if(T.get_virtual_z_level() != affected_mob.get_virtual_z_level())
+			continue
+		if(T.name == "space")
+			continue
+		if(!T.density)
+			var/clear = 1
+			for(var/obj/O in T)
+				if(O.density)
+					clear = 0
+					break
+			if(clear)
+				L+=T
```

Comment:
Use early returns instead of embedding the ifs
```
if(T.density)
  continue
```

Reply:
👍